resources/metadata/cnbBuild.yaml

```diff
@@ -135,6 +135,34 @@ spec:
           - PARAMETERS
           - STAGES
           - STEPS
+      - name: bindings
+        type: "map[string]interface{}"
+        description: |
+          Map of bindings that should be offered to the buildpack. The type of bindings depend on the buildpack. For documentation about bindings in general see [the paketo documentation](https://paketo.io/docs/howto/configuration/#bindings).
+
+          Example: Custom maven settings.xml for the Java Buildpack
+
+          ```yaml
+          bindings:
+            maven-settings:
+              type: maven
+              secret: settings.xml
+              file: path/to/settings.xml
+          ```
+
+          or inline
+
+          ```yaml
+          bindings:
+            maven-settings:
+              type: maven
+              secret: settings.xml
+              content: "inline settings.xml"
```

Comment:
Will all binding contents be strings or should we support encoded data like Kubernetes Secrets do?

Reply:
> Will all binding contents be strings or should we support encoded data like Kubernetes Secrets do?

I think `secret` is rather misleading. But how would the usage with Kubernetes Secrets look like?

Reply:
What I meant was that Kubernetes Secrets' contents are base64 encoded because the support any binary content while resources only support character data.

Reply:
If you have non utf-8 content, you can go for `file` instead of `content`.

Therefore, I would not require base64 encoding, because it will just make it worse for everyone with only utf-8 data (writing and reviewing).

If this is really an important use-case, we can later always add a `base64Content` alternative.

Agree?